### PR TITLE
Stream verified block

### LINKED ISSUE #8198: StreamBlock RPC streams unverified block
# 💎 Issue

`StreamBlock` RPC end point currently streams not-processed and not-verified block. Similar to rest of our RPC unary and stream end points, a beacon node should verify the beacon object before sending it out.  Sending unverified object is generally bad as it requires downstream services to verify them while most of them won't be able to unless downstream services maintain its own beacon state which is expensive. The `StreamBlock` also has a bug, it uses head state to verify block signature, it should have used parent state

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

Implement `StreamBlock` to support streaming of verified block instead of just unverified block that is received via p2p. To achieve this with simplicity, we added a new field `SignedBlock` to `BlockProcessedData` which is used by the event feed

**Which issues(s) does this PR fix?**

Fixes #8198

**Other notes for review**

### CODE REVIEW

**Review by rauljordan — DISMISSED** [review]

File: endtoend/evaluators/slashing.go [hunk 1]
```diff
@@ -35,14 +35,14 @@ var ProposeDoubleBlock = types.Evaluator{
 // ValidatorsSlashed ensures the expected amount of validators are slashed.
 var ValidatorsSlashed = types.Evaluator{
 	Name:       "validators_slashed_epoch_%d",
-	Policy:     policies.AfterNthEpoch(1),
+	Policy:     policies.AfterNthEpoch(2),
```
Comment (prestonvanloon) [inline endtoend/evaluators/slashing.go:38]: Why are there changes here?

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Please explain e2e changes.

**Comment by terencechain** [comment]
Did more research. Stream unverified block is needed and only used by slasher. Until we revamp slasher, we'll unfortunately have to keep this method. I'll go ahead and implement a new RPC method that is called `StreamVerifiedBlock`

File: endtoend/evaluators/slashing.go (hunk 1 above)
Comment (terencechain) [reply endtoend/evaluators/slashing.go:38]: Sorry I was doing some local testing. Didn't mean to push this. Will revert it back

File: beacon-chain/rpc/beacon/blocks.go [hunk 2]
```diff
@@ -175,35 +176,53 @@ func (bs *Server) GetChainHead(ctx context.Context, _ *ptypes.Empty) (*ethpb.Cha
...
-				}
-				if err := stream.Send(data.SignedBlock); err != nil {
-					return status.Errorf(codes.Unavailable, "Could not send over stream: %v", err)
+					if err := blocks.VerifyBlockSignature(headState, data.SignedBlock); err != nil {
+						log.WithError(err).WithField("blockSlot", data.SignedBlock.Block.Slot).Error("Could not verify block signature")
+						continue
+					}
+					if err := stream.Send(data.SignedBlock); err != nil {
+						return status.Errorf(codes.Unavailable, "Could not send over stream: %v", err)
+					}
```
Comment (terencechain) [inline beacon-chain/rpc/beacon/blocks.go:225]: these are old code, we just refactored it under the `else`

**Review by prestonvanloon — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**